collections.md

```diff
@@ -167,3 +168,30 @@ Collection::times(7, function ($number) {
 
 Tip given by [@Teacoders](https://twitter.com/Teacoders/status/1509447909602906116)
 
+### Collection Paginate
+
+Since laravel 5.1, you can create your own paginate.
+
+```php
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Support\Collection;
+
+public static function collectionPaginate(Collection $collection, int $perPage, int $currentPage): LengthAwarePaginator
+{
+    $perPage = max($perPage, 1);
+    $totalPages = ceil($collection->count() / $perPage);
+    $currentPage = min($totalPages, max($currentPage, 1));
+
+    $skip = ($currentPage - 1) * $perPage;
+    $take = min($collection->count(), $perPage);
+
+    $collection = $collection->skip($skip)
+        ->take($take)
+        ->values();
+    return new LengthAwarePaginator($collection, $totalPages, $perPage);
```

Comment:
```suggestion
    $perPage = max($perPage, 1);
    $totalCount = $collection->count();
    $totalPages = ceil($totalCount / $perPage);
    $currentPage = min($totalPages, max($currentPage, 1));

    return new LengthAwarePaginator(
        $collection->forPage($currentPage, $perPage)
        $totalCount,
        $perPage,
        $currentPage
    );
```

The docs suggest slicing the collection and provide a method for this purpose called `forPage()` (existed in 5.1, possibly earlier).

The Laravel API reference indicates that the second parameter of LengthAwarePaginator is supposed to be the `The total number of items before slicing.` and not the total page count.